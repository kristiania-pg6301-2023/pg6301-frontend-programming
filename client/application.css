body {
  background-color: rebeccapurple;
  margin: 0;
}

#app {
  height: 100vh;
  display: grid;
  grid-template-rows: auto auto 1fr auto;
}

main {
  background-color: aliceblue;
  min-width: 80%;
  margin-inline: auto;
  margin-block: 1em;
  border-radius: 1em;
}

header,
nav,
footer {
  color: aliceblue;
}

nav a {
  color: aliceblue;
}
