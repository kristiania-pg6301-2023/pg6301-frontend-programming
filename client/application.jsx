import React, {useEffect, useRef, useState} from "react";

function ListTasks() {
    const [tasks, setTasks] = useState(undefined);

    useEffect(() => {
        async function fetchData() {
            const res = await fetch("/api/todos");
            setTasks(await res.json());
        }

        fetchData();
    }, [])

    if (!tasks) {
        return <div>Loading ...</div>;
    }

    return <>
        {tasks.map(t => <div key={t.title}>{t.title}</div>)}
    </>;
}

function AddTaskButton() {
    const dialogRef = useRef();
    const [showDialog, setShowDialog] = useState(false);
    useEffect(() => {
        if (showDialog) {
            dialogRef.current.showModal();
        } else {
            dialogRef.current.close();
        }
    }, [showDialog]);
    const [title, setTitle] = useState("");

    async function handleSubmit() {

    }

    return <>
        <dialog ref={dialogRef}>
            <h2>Add task</h2>
            <form onSubmit={handleSubmit} method={"dialog"}>
                <div>
                    Title <br />
                    <input autoFocus={true} value={title} onChange={e => setTitle(e.target.value)}/>
                </div>
                <div>
                    <button>Submit</button>
                    <button onClick={() => setShowDialog(false)}>Close</button>
                </div>
            </form>
        </dialog>
        <button onClick={() => setShowDialog(true)}>Add task</button>
    </>;
}


export function TodoApplication() {
    return <>
        <h1>Welcome to task application</h1>

        <h2>Tasks</h2>
        <AddTaskButton/>
        <ListTasks/>

    </>;
}