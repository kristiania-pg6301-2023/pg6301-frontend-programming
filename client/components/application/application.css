* {
  margin: 0;
}

#root {
  min-height: 100vh;
  display: grid;
  grid-template-rows: auto auto 1fr auto;
  background-color: rebeccapurple;
  color: white;
}

a {
  color: white;
}

header,
nav,
main,
footer {
  padding: 0.25em 1em;
}

main {
  background-color: aliceblue;
  color: #101010;
  min-width: 80%;
  margin: 0.5em auto;
}

main a {
  color: #101010;
}
