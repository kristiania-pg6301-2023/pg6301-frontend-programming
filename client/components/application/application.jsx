import { useEffect, useState } from "react";
import { LoginButton } from "../login/loginButton";
import { fetchJSON } from "../../lib/fetchJSON";
import { LoginCallback } from "../login/loginCallback";

export function Application() {
  async function loadConfig() {
    setApplicationConfig(await fetchJSON("/api/config"));
  }

  const [applicationConfig, setApplicationConfig] = useState();
  useEffect(() => {
    loadConfig();
  }, []);

  if (!applicationConfig) {
    return <div>Loading...</div>;
  }

  if (window.location.pathname === "/login/callback") {
    return <LoginCallback applicationConfig={applicationConfig} />;
  }

  if (!applicationConfig.user) {
    return <LoginButton applicationConfig={applicationConfig} />;
  }

  return <h1>Application</h1>;
}
