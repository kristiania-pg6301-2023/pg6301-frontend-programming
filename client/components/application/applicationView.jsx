export function ApplicationView({ user }) {
  return <h1>Welcome {user.name} ({user.email})</h1>;
}