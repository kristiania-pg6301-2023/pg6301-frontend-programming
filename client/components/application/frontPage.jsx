import { Link } from "react-router-dom";

export function FrontPage() {
  return (
    <>
      <h2>Front page</h2>
      <ul>
        <li>
          <Link to={"/movies"}>List movies</Link>
        </li>
        <li>
          <Link to={"/movies/new"}>Add new movie</Link>
        </li>
      </ul>
    </>
  );
}
