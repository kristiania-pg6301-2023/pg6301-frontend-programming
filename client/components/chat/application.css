* {
  margin: 0;
}

#root {
  height: 100vh;
  display: grid;
  grid-template-rows: auto 1fr auto;
}

header {
  background-color: rebeccapurple;
  color: aliceblue;
}

main > div {
  border: 1px solid rebeccapurple;
}

main {
  overflow-y: scroll;
}

footer form {
  display: flex;
}

footer form input {
  flex: 1;
}
