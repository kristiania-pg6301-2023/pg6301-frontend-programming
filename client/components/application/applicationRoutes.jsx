import { Route, Routes } from "react-router-dom";
import { FrontPage } from "./frontPage";
import { MoviesList } from "../movies/moviesList";
import { AddMovieForm } from "../movies/addMovieForm";
import { useEffect, useState } from "react";

const MOVIES = [
  {
    _id: "1",
    title: "Barbie",
    year: 2023,
    countries: ["USA"],
  },
  {
    _id: "2",
    title: "Oppenheimer",
    year: 2023,
    countries: ["USA"],
  },
];

export function ApplicationRoutes() {
  const [movies, setMovies] = useState([]);

  async function loadMovies() {
    setMovies(MOVIES);
  }

  useEffect(() => {
    loadMovies();
  }, []);

  return (
    <Routes>
      <Route path={"/"} element={<FrontPage />} />
      <Route path={"/movies"} element={<MoviesList movies={movies} />} />
      <Route path={"/movies/new"} element={<AddMovieForm />} />
      <Route path={"*"} element={<h2>Not Found</h2>} />
    </Routes>
  );
}
