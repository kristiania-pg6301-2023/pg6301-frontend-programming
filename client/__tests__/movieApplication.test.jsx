import renderer from "react-test-renderer";
import {ListMovies} from "../application";

describe("the movie application", () => {

    it("shows movie list", () => {
        const movieList = [
            {title: "Oppenheimer", year: "2023", plot: "Manhattan project"},
            {title: "Barbie", year: "2023", plot: "Plastic adventure"},
        ];
        const component = renderer.create(<ListMovies movies={movieList} />);
        expect(component.root.findAllByType("h3")[0].children).toEqual([
            "Oppenheimer", " (", "2023", ")"
        ]);
        expect(component).toMatchSnapshot();
    })

});