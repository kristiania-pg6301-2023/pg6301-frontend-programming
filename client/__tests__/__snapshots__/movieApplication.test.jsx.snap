// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the movie application shows movie list 1`] = `
[
  <h1>
    List Movies
  </h1>,
  <div>
    <h3>
      Oppenheimer
       (
      2023
      )
    </h3>
    <div>
      Manhattan project
    </div>
  </div>,
  <div>
    <h3>
      Barbie
       (
      2023
      )
    </h3>
    <div>
      Plastic adventure
    </div>
  </div>,
]
`;
